src/fswAlgorithms/attDetermination/inertialAttitudeUkf/tests/test_inertialAttitudeUkf.py

```diff
@@ -0,0 +1,425 @@
+    x = np.zeros([len(t), len(x0) + 1])
+    h = (t[len(t) - 1] - t[0]) / len(t)
+    x[0, 0] = t[0]
+    x[0, 1:] = x0
+    for i in range(len(t) - 1):
+        h = t[i + 1] - t[i]
+        x[i, 0] = t[i]
+        k1 = h * f(t[i], x[i, 1:], Inertia)
+        k2 = h * f(t[i] + 0.5 * h, x[i, 1:] + 0.5 * k1, Inertia)
+        k3 = h * f(t[i] + 0.5 * h, x[i, 1:] + 0.5 * k2, Inertia)
+        k4 = h * f(t[i] + h, x[i, 1:] + k3, Inertia)
+        x[i + 1, 1:] = x[i, 1:] + (k1 + 2. * k2 + 2. * k3 + k4) / 6.
+        if mrpShadow:
+            s = np.linalg.norm(x[i + 1, 1:4])**2
+            if s > 1:
+                x[i + 1, 1:4] = - (x[i + 1, 1:4]) / s
+        x[i + 1, 0] = t[i + 1]
+    return x
+
+def attitude_dynamics(t, x, I):
+    dxdt = np.zeros(np.shape(x))
+    mrp = x[:3]
+    omega = x[3:]
+    B = rbk.BmatMRP(mrp)
+    dxdt[:3] = 0.25 * np.matmul(B, omega)
+    dxdt[3:] = - np.dot(np.linalg.inv(I), np.cross(omega, np.dot(I, omega)))
+    return dxdt
+
+def mrp_integration(t, x, I=np.eye(3)):
+    dxdt = np.zeros(np.shape(x))
+    B = rbk.BmatMRP(x[0:3])
+    dxdt[:3] = 0.25 * np.matmul(B, x[3:])
+    dxdt[3:] = np.zeros(3)
+    return dxdt
+
+
+def setup_filter_data(filter_object):
+    filter_object.setAlpha(0.02)
+    filter_object.setBeta(2.0)
+
+    states = [0.05, 0.5, 0.1, 0.2, -0.05, 0.1]
+
+    filter_object.setInitialState([[s] for s in states])
+    filter_object.setInitialCovariance([[1, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                        [0.0, 1, 0.0, 0.0, 0.0, 0.0],
+                                        [0.0, 0.0, 1, 0.0, 0.0, 0.0],
+                                        [0.0, 0.0, 0.0, 0.01, 0.0, 0.0],
+                                        [0.0, 0.0, 0.0, 0.0, 0.01, 0.0],
+                                        [0.0, 0.0, 0.0, 0.0, 0.0, 0.01]])
+    filter_object.setGyroNoise([[1e-3,0,0],[0.,1e-3,0],[0,0,1e-3]])
+    sigma_mrp = (1E-2) ** 2
+    sigma_rate = (1E-2) ** 2
+    filter_object.setProcessNoise([[sigma_mrp, 0.0, 0.0, 0.0, 0.0, 0.0],
+                                   [0.0, sigma_mrp, 0.0, 0.0, 0.0, 0.0],
+                                   [0.0, 0.0, sigma_mrp, 0.0, 0.0, 0.0],
+                                   [0.0, 0.0, 0.0, sigma_rate, 0.0, 0.0],
+                                   [0.0, 0.0, 0.0, 0.0, sigma_rate, 0.0],
+                                   [0.0, 0.0, 0.0, 0.0, 0.0, sigma_rate]])
+
+    filter_object.setLowPassFilter(0.5, 15/(2*np.pi))
+
+    return
+
+@pytest.mark.parametrize("show_plots", [False])
+def test_propagation_kf(show_plots):
+    """Module Unit Test"""
+    state_propagation_flyby(show_plots)
+
+@pytest.mark.parametrize("show_plots", [False])
+@pytest.mark.parametrize("initial_error", [False, True])
+@pytest.mark.parametrize("method", [starOnly, gyroWhenDazzled, allMeasurements])
+def test_measurements_kf(show_plots, initial_error, method):
+    """Module Unit Test"""
+    state_update_flyby(show_plots, initial_error, method)
```

Comment:
:large_orange_diamond: This indirection can be removed. Rename the called function `test_` etc.

Same for the above test function

Reply:
done